src/server/pps/server/commit_set.go

```diff
@@ -79,19 +60,72 @@ func newBranchSetFactory(ctx context.Context, pfsClient pfs.APIClient, inputs []
 					commitInfo, err := stream.Recv()
 					if err != nil {
 						select {
+						case <-ctx.Done():
 						case errCh <- err:
-						default:
 						}
 						return
 					}
-					branchCh <- &pfs.Branch{
+					select {
+					case <-ctx.Done():
+						return
+					case branchCh <- &pfs.Branch{
 						Name: branchName,
 						Head: commitInfo.Commit,
+					}:
 					}
 				}
 			}(branchName)
 		}
 	}
 
+	ch := make(chan *branchSet)
```

Comment:
Just to make sure I understand this code: is this part of the scaling change, or just more general refactoring?